clang/lib/Driver/Driver.cpp

```diff
@@ -1023,20 +1023,19 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
 
   llvm::SmallVector<llvm::Triple, 4> UniqueSYCLTriplesVec;
 
-  // If -fsycl is supplied without -fsycl-targets we will assume SPIR-V.
-  // For -fsycl-device-only, we also setup the implied triple as needed.
-  if (HasValidSYCLRuntime) {
+  // If -fsycl is supplied we will assume SPIR-V.
```

Comment:
I am a bit confused. How can we assume SPIR-V for -fsycl. What about -fsycl -fsycl-targets=nvptx?

Reply:
`-fsycl` by itself has always implied SPIR-V.  This initial support is for JIT only, which means not introducing `-fsycl-targets` yet.

Reply:
I would phrase this differently. `-fsycl` implies that device code is compiled for some target. The target for device code is set by `--offload-arch` (i.e. `-fsycl` implies `--offload-arch`). DPC++ uses SPIR-V by default. It should be reasonable default for upstream too, unless community sets some other default value for `--offload-arch` already.